### PR TITLE
[1.18] build: Update Debian base images

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/area dependency security
/sig release api-machinery

**What this PR does / why we need it**:

- build: Update Debian base images
  - debian-base:v2.1.3
  - debian-iptables:v12.1.2
  - debian-hyperkube-base:v1.1.3
- build: Remove Debian base image building

/assign @BenTheElder 
cc: @kubernetes/release-engineering @cjcullen 
ref: https://github.com/kubernetes/release/pull/1450, https://github.com/kubernetes/kubernetes/pull/93562, https://github.com/kubernetes/kubernetes/issues/92616, https://github.com/kubernetes/kubernetes/issues/93114

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Cherry pick of https://github.com/kubernetes/kubernetes/pull/93916.

**Does this PR introduce a user-facing change?**:

```release-note
- build: Update Debian base images
  - debian-base:v2.1.3
  - debian-iptables:v12.1.2
  - debian-hyperkube-base:v1.1.3
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: build/OWNERS [hunk 1]
```diff
@@ -11,6 +11,7 @@ approvers:
   - bentheelder
   - cblecker
   - fejta
+  - justaugustus
```
Comment (dims) [inline build/OWNERS:14]: Can you please split this into another PR?

**Comment by dims** [comment]
@justaugustus let's leave the OWNERS file changes to another PR?

**Review by saschagrunert — APPROVED** [review]
/lgtm

File: build/OWNERS (hunk 1 above)
Comment (justaugustus) [reply build/OWNERS]: Removed

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm
